packages/@react-spectrum/s2/src/Disclosure.tsx

```diff
@@ -195,7 +201,52 @@ const chevronStyles = style({
   flexShrink: 0
 });
 
+
 function DisclosureHeader(props: DisclosureHeaderProps, ref: DOMRef<HTMLDivElement>) {
+  let {
+    UNSAFE_className,
+    UNSAFE_style,
+    children
+  } = props;
+  let domRef = useDOMRef(ref);
+  let {size, isQuiet, isDisabled, density} = useSlottedContext(DisclosureContext)!;
+
+
+  let mapSize = {
+    S: 'XS',
+    M: 'S',
+    L: 'M',
+    XL: 'L'
+  };
+
+  // maps to one size smaller in the compact density to ensure there is space between the top and bottom of the action button and container
+  let newSize : 'XS' | 'S' | 'M' | 'L' | 'XL' | undefined = size;
+  if (density === 'compact') {
+    newSize = mapSize[size ?? 'M'] as 'XS' | 'S' | 'M' | 'L';
+  }
+
+  return (
+    <Provider
+      values={[
+        [ActionButtonContext, {size: newSize, isQuiet, isDisabled}]
```

Comment:
ActionButton isn't disabled for some reason. 
<img width="245" alt="image" src="https://github.com/user-attachments/assets/23f26de8-ab55-454e-98b8-bb63c57532fa">

Reply:
ah yeah, that's because isDisabled doesn't get passed to the DisclosureContext from Accordion. that said, from testing yesterday, i believe we said that we don't want to disable the action button even when disclosure is disabled. if they want to disable it, they can pass disable directly to the component. 

im not sure if that's the same behavior we want to keep when the whole accordion is disabled though. thoughts?